spark2/src/main/java/org/apache/iceberg/spark/source/Reader.java

```diff
@@ -180,6 +184,8 @@ private Expression filterExpression() {
 
   private StructType lazyType() {
     if (type == null) {
+      Preconditions.checkArgument(readTimestampWithoutZone || !SparkUtil.hasTimestampWithoutZone(lazySchema()),
```

Comment:
Actually cancel what I said about those other spots, this seems like a great place to check for whether we are allowed to do the TZ conversion

Reply:
ok